conda/plugins/manager.py

```diff
@@ -181,6 +181,14 @@ def yield_command_hook_actions(self, hook_type: CommandHookTypes, command: str):
             if command in command_hook.run_for:
                 yield command_hook.action
 
+    def disable_external_plugins(self) -> None:
+        """
+        Disables all currently registered plugins except built-in conda plugins
+        """
+        for name, _ in self.list_name_plugin():
+            if not name.startswith("conda."):
+                self.set_blocked(name)
```

Comment:
You're right, the name is created by https://github.com/pytest-dev/pluggy/blob/1158d387b7622a9b4d66bf8bcdc1c38effe9f497/src/pluggy/_manager.py#L255-L264 only during registration if it's not provided, e.g. when we load internal plugin, then the name will be the dotted Python path to the module of the plugin.

But for plugins provided by 3rd party packages via entrypoints, it goes through the [`CondaPluginManager.load_entrypoints`](https://github.com/conda/conda/blob/main/conda/plugins/manager.py#L67-L99) which will use the name of the entrypoint, e.g. here's the output on my machine:

```
>>> pprint.pprint(context.plugin_manager.list_name_plugin())
[('conda.plugins.solvers',
  <module 'conda.plugins.solvers' from '/path/to/conda/conda/plugins/solvers.py'>),
 ('conda.plugins.virtual_packages.archspec',
  <module 'conda.plugins.virtual_packages.archspec' from '/path/to/conda/conda/plugins/virtual_packages/archspec.py'>),
 ('conda.plugins.virtual_packages.cuda',
  <module 'conda.plugins.virtual_packages.cuda' from '/path/to/conda/conda/plugins/virtual_packages/cuda.py'>),
 ('conda.plugins.virtual_packages.freebsd',
  <module 'conda.plugins.virtual_packages.freebsd' from '/path/to/conda/conda/plugins/virtual_packages/freebsd.py'>),
 ('conda.plugins.virtual_packages.linux',
  <module 'conda.plugins.virtual_packages.linux' from '/path/to/conda/conda/plugins/virtual_packages/linux.py'>),
 ('conda.plugins.virtual_packages.osx',
  <module 'conda.plugins.virtual_packages.osx' from '/path/to/conda/conda/plugins/virtual_packages/osx.py'>),
 ('conda.plugins.virtual_packages.windows',
  <module 'conda.plugins.virtual_packages.windows' from '/path/to/conda/conda/plugins/virtual_packages/windows.py'>),
 ('conda.plugins.subcommands.doctor.cli',
  <module 'conda.plugins.subcommands.doctor.cli' from '/path/to/conda/conda/plugins/subcommands/doctor/cli.py'>),
 ('conda-libmamba-solver',
  <module 'conda_libmamba_solver.plugin' from '/path/to/miniconda/lib/python3.10/site-packages/conda_libmamba_solver/plugin.py'>)]
```

See the `conda-libmamba-solver` at the end? That's the name of the entrypoint provided by conda plugin authors.

I would suggest adding a condition to `load_entrypoints` that would check if the entrypoint name is not starting with `conda.` and continue otherwise.

Reply:
To clarify, basically we should be looking into making sure that when we load plugins, from 3rd party apps, that we can successfully disable them, independent of what the authors have provided as name.

There are a few ways to do that, one of which could be that we mark all built-in plugins with a name prefix such as `conda:` or something, but verify that the loaded plugin module is actually inside the `conda` package namespace as well (during disabling?). Then, when we want to disable any 3rd party plugins, we can simply disable any that don't have the `conda:` prefix in the name.

Reply:
My first thought was that since `self.list_name_plugin()` tells us the `(name: str, plugin: module)`, we can just inspect the `plugin` itself and ensure it's loaded from the `conda` module. But now I'm wondering if this too could be spoofed via a [packaging namespace](https://packaging.python.org/en/latest/guides/packaging-namespace-packages/)?

Reply:
I've tweaked the `disable_external_plugins` method a little to account for module and class based plugins and fallback to automatically disable the plugin if it can't be identified.

Reply:
In [`6707f1e` (#12824)](https://github.com/conda/conda/pull/12824/commits/6707f1e3509691e0fb86bf127a120c9337a78431) I mean